cmd/s3-peer-client.go

```diff
@@ -19,125 +19,74 @@ package cmd
 import (
 	"encoding/json"
 	"fmt"
-	"net/rpc"
 	"net/url"
 	"path"
 	"sync"
-	"time"
 )
 
 type s3Peers struct {
-	// A map of peer server address (in `host:port` format) to RPC
-	// client connections.
-	rpcClients map[string]*AuthRPCClient
+	// A map of peer server address (in `host:port` format) to
+	// BucketMetaState interface
+	bmsClients map[string]BucketMetaState
 
-	mutex *sync.RWMutex
-
-	// Is single-node?
-	isDistXL bool
+	// Is distributed?
+	IsDistXL bool
 
 	// Slice of all peer addresses (in `host:port` format).
-	peers []string
+	Peers []string
 }
 
-func initGlobalS3Peers(eps []*url.URL) {
-	// Get list of de-duplicated peers.
-	peers := getAllPeers(eps)
+func mkPeersFromEndpoints(eps []*url.URL) s3Peers {
+	bmsc := make(map[string]BucketMetaState)
 
-	// Initialize global state.
-	globalS3Peers = s3Peers{
-		rpcClients: make(map[string]*AuthRPCClient),
-		mutex:      &sync.RWMutex{},
-	}
+	// Setup local server state change handler
+	bmsc[globalMinioAddr] = &localBMS{ObjectAPI: newObjectLayerFn}
 
-	// Initialize each peer connection.
-	for _, peer := range peers {
-		globalS3Peers.InitS3PeerClient(peer)
-	}
+	peers := []string{globalMinioAddr}
 
-	// Save new peers
-	globalS3Peers.peers = peers
-
-	// store if this is a distributed setup or not.
-	globalS3Peers.isDistXL = len(globalS3Peers.peers) > 1
-}
+	for _, ep := range eps {
+		if ep == nil || ep.Host == "" || ep.Host == globalMinioAddr {
```

Comment:
Done. Also removed `ep.Host == globalMinioAddr` as it is just making the code look more complex but this case is handled later in the loop.